src/hdf5/HDF5IO.cpp

```diff
@@ -377,6 +379,21 @@ Status HDF5IO::createStringDataSet(const std::string& path,
   return Status::Success;
 }
 
+Status HDF5IO::startRecording()
+{
+  modifyObjects = false;
+  if (H5Fstart_swmr_write(this->file->getId()) < 0) {
+    return Status::Failure;
+  }
+  return Status::Success;
+}
+
+Status HDF5IO::stopRecording()
+{
+  modifyObjects = true;
+  return Status::Success;
```

Comment:
If I understand it correctly, this doesn't actually stop the `SWMR` mode, i.e., even though `modifyObjects` is set to `true` here, modifying objects would still fail. I don't think there is a `H5Fstop_swmr_write` function, so I think what we need to do here is:

- flush all data to disk
- close the HDF5 file
- open the HDF5 file again in `r+` mode

Reply:
Good point. Yes, I don't think there is a `H5Fstop_swmr_write` function, the recommended usage is to close and open again in `r+` mode.

We discussed that there is likely a distinction between pausing and stopping a recording (https://github.com/NeurodataWithoutBorders/aqnwb/issues/28), but I'm not sure yet what logic we would want to implement for each. I think it makes sense for now for `stopRecording` to close / reopen the file, and we can think about any distinctions for `pauseRecording` when I address that issue.

Reply:
> We discussed that there is likely a distinction between pausing and stopping a recording (#28), but I'm not sure yet what logic we would want to implement for each.

I think the current version of the code already implements this in `NWBRecording` in that:

* `NWBFile.stopRecording` means to stop the recording process, but leave the file open (so that a user can continue later on)
* `NWBFile.finalize` means to stop the recording and close the file (i.e., a true stop)

In that sense, we could rename `stopRecording` to `pauseRecording` in both `NWBFile` and `baseIO` if that is more intuitive.

Reply:
> think the current version of the code already implements this in NWBRecording

That's true. I changed the name to `pauseRecording`. 

I also added a test to check that the `recordingContainers` time series are still accessible after pausing the recording - I think that would be the other main difference between stopping and pausing?

Reply:
On second thought - I'm not sure they should still be accessible, and I think this might be preventing the file from actually closing properly? I'm running into issues reopening the file during the `pauseRecording` function without `H5F_ACC_SWMR_WRITE` intent, and I think this may be the issue.

Reply:
> I also added a test to check that the `recordingContainers` time series are still accessible after pausing the recording - I think that would be the other main difference between stopping and pausing?

Yeah, I think that's a problem. When closing and reopening the file, I believe the recordingContainers are all going to be invalid, because the HDF5 objects they point to depend on the file object that was opened before. The only way to really prevent this from happending would be if the recordingContainers (or any NWB object) would not store the object but request them from the IO object each time. Let's discuss tomorrow how we want to address this. Maybe we should just leave it as a true stop of the recording for now, i.e., no restart. Once we add read, we could then do a restart by opening the file again.

Reply:
Other than this last detail, the PR looks good.